web/modules/custom/joinup_discussion/src/EventSubscriber/WorkflowTransitionEventSubscriber.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\joinup_discussion\EventSubscriber;
+
+use Drupal\comment\Plugin\Field\FieldType\CommentItemInterface;
+use Drupal\state_machine\Event\WorkflowTransitionEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Event subscriber to close comments on archived discussions.
```

Comment:
Action is called disable, state is called archived to uniform with the rest of the states. What are we going to use?